PR: https://github.com/odoo/odoo/pull/38424

From: 9ba8734f15e1a292ca27b1a026e8366a91b2a8c9
From: wan
Date: 2019-10-18 12:53:41

Structural Changes: 1
Total Changes: 38

[MOV] account: reconciliation widget
Task 2034073
The reconciliation widget is now entirely part of the enterprise version
as it is an advanced accounting feature

closes odoo/odoo#38424

Signed-off-by: Quentin De Paoli (qdp) <qdp@openerp.com>

================================= pseudo patch: =================================

--- a/addons/stock_account/models/__init__.py
+++ b/addons/stock_account/models/__init__.py
@@ -3,7 +3,6 @@
 
 from . import account_chart_template
 from . import account_move
-from . import account_reconciliation_widget
 from . import product
 from . import stock_move
 from . import stock_inventory

--- a/addons/stock_account/models/account_reconciliation_widget.py
+++ b/None
@@ -1,37 +0,0 @@
-# -*- coding: utf-8 -*-
-
-from odoo import api, models
-from odoo.osv import expression
-
-
-class AccountReconciliation(models.AbstractModel):
-    _inherit = "account.reconciliation.widget"
-
-    @api.model
-    def _domain_move_lines_for_reconciliation(self, st_line, aml_accounts, partner_id, excluded_ids=None, search_str=False, mode='rp'):
-        def to_int(val):
-            try:
-                return int(val)
-            except (ValueError, TypeError):
-                return None
-
-        domain = super()._domain_move_lines_for_reconciliation(
-            st_line, aml_accounts, partner_id, excluded_ids=excluded_ids, search_str=search_str, mode=mode
-        )
-        acc_props = (
-            "property_stock_account_input",
-            "property_stock_account_output",
-            "property_stock_account_input_categ_id",
-            "property_stock_account_output_categ_id",
-        )
-        acc_ids = [
-            (acc["value_reference"] or "").split(",")[-1]
-            for acc in self.env["ir.property"]
-            .sudo()
-            .search([("name", "in", acc_props), ("value_reference", "!=", False)])
-            .read(["value_reference"])
-            if to_int((acc["value_reference"] or "").split(",")[-1])
-        ]
-        if acc_ids:
-            domain = expression.AND([domain, [("account_id.id", "not in", acc_ids)]])
-        return domain
